test/unit/url_to_subscriber_list_migration_test.rb

```diff
@@ -0,0 +1,211 @@
+    output = run_migration_on(csv_data, static_data, "true")
+    expected_output = <<-STR.strip_heredoc
+      .
+      1 parsed
+      Skipped: {}
+    STR
+    assert_equal expected_output, output
+  end
+
+  test "Correctly look up values using `UrlToSubscriberListCriteria::StaticData` class" do
+    database_lookup_test(UrlToSubscriberListCriteria::StaticData)
+  end
+
+  test "Correctly look up values using `UrlToSubscriberListCriteria::BulkStaticData` class" do
+    database_lookup_test(UrlToSubscriberListCriteria::BulkStaticData.new)
+  end
+
+  def database_lookup_test(static_data)
+    url = "https://www.gov.uk/government/publications.atom" +
+      "?departments%5B%5D=academy-for-justice-commissioning" +
+      "&world_locations%5B%5D=united-kingdom" +
+      "&topics%5B%5D=defence-and-armed-forces"
+
+    url_people = "https://www.gov.uk/government/people/david-cameron.atom"
+    url_role = "https://www.gov.uk/government/ministers/chancellor-of-the-exchequer.atom"
+    url_topic_event = "https://www.gov.uk/government/topical-events/autumn-statement-2016.atom"
+
+    csv_data = [
+      { topic_id: 'TOPIC_A', url: url, created: Time.zone.now },
+      { topic_id: 'TOPIC_B', url: url_people, created: Time.zone.now },
+      { topic_id: 'TOPIC_C', url: url_role, created: Time.zone.now },
+      { topic_id: 'TOPIC_D', url: url_topic_event, created: Time.zone.now },
+    ]
+
+    csv_data.each do |row|
+      stub_request(:get, %r{email-alert-api.test.alphagov.co.uk/subscriber-lists}).to_return(
+        body: {'subscriber_list' => { 'gov_delivery_id' => row[:topic_id], updated_at: '2011-11-11' } }.to_json
+      )
+    end
+
+    organisation = create(:organisation, slug: 'academy-for-justice-commissioning')
+    world_location = create(:world_location, slug: 'united-kingdom')
+    policy_area = create(:topic, slug: 'defence-and-armed-forces')
+    person = create(:person, slug: 'david-cameron')
+    role = create(:role, slug: 'chancellor-of-the-exchequer')
+    topical_event = create(:topical_event, slug: 'autumn-statement-2016')
+
+    output = run_migration_on(csv_data, static_data)
+    expected_output = <<-STR.strip_heredoc
+      ******* GovUkDelivery details
+      Parsing TOPIC_A - https://www.gov.uk/government/publications.atom?departments%5B%5D=academy-for-justice-commissioning&world_locations%5B%5D=united-kingdom&topics%5B%5D=defence-and-armed-forces
+      ******* Converted Hash values
+      {"links"=>{"world_locations"=>["united-kingdom"], "organisations"=>["academy-for-justice-commissioning"], "policy_areas"=>["defence-and-armed-forces"]}, "email_document_supertype"=>"publications"}
+      {"links"=>{"world_locations"=>["#{world_location.content_id}"], "organisations"=>["#{organisation.content_id}"], "policy_areas"=>["#{policy_area.content_id}"]}, "email_document_supertype"=>"publications"}
+      ******* EmailAlertApi details
+      {"gov_delivery_id"=>"TOPIC_D", "updated_at"=>"2011-11-11"}
```

Comment:
opps.. that's my fault I did the stubbing incorrectly, had meant it to have a value that matched for each row.